dev/run-tests.py

```diff
@@ -333,7 +333,7 @@ def build_spark_assembly_sbt(hadoop_version, checkstyle=False):
     exec_sbt(profiles_and_goals)
 
     if checkstyle:
-        run_java_style_checks()
+        run_java_style_checks(" ".join(build_profiles))
```

Comment:
Let's pass `build_profiles` as is and join the strings at `run_java_style_checks`.